7_react/react1/week3/todo-app/src/component/TodoContainer.jsx

```diff
@@ -0,0 +1,132 @@
+                </span>
+            </div>
+        </div>
+    )
+}
+
+const TodoContainer = ({ jsonTodos }) => {
+  const [newTodo, setNewTodo] = useState('');
+  const [todos, setTodos] = useState([]);
+  const [selectedDate, setDate] = useState(new Date());
+
+  useEffect(() => {
+    setTodos(jsonTodos);
+  }, [jsonTodos]);
+
+  const handleAddTodo = () => {
+    if (newTodo.trim() !== '' ) {      
+        const newTodoItem = {
+        id: todos.length + 1, 
+        title: newTodo,
+        deadline:selectedDate,
+        completed: false
+      };
+      setTodos([...todos, newTodoItem]);
+      setNewTodo('');
+      
+    }else {
+        alert('Please create new items & deadline for the todo.');
+      }
+  };
+
+  const handleEdit = (id) => {
+    const updatedTodos = todos.map(todo =>
+      todo.id === id ? { ...todo, editing: true } : todo
+    );
+    setTodos(updatedTodos);
+  };
+
+  const handleEditChange = (e, id) => {
+    const updatedTodos = todos.map(todo =>
+      todo.id === id ? { ...todo, title: e.target.value } : todo
+    );
+    setTodos(updatedTodos);
+  };
+
+  const handleUpdate = (id) => {
+    const updatedTodos = todos.map(todo =>
+      todo.id === id ? { ...todo, editing: false } : todo
+    );
+    setTodos(updatedTodos);
+  };
+
+  const handleDelete = (id) => {
+    const updatedTodos = todos.filter(todo => todo.id !== id);
+    setTodos(updatedTodos);
+  };
+
+   const handleCheckboxChange = (id) => {
+    const updatedTodos = todos.map(todo =>
+      todo.id === id ? { ...todo, completed: !todo.completed } : todo
+    );
+    setTodos(updatedTodos);
+  };
+  
+
+
+  return (
+    <div className="todo-container"style={{backgroundColor: "#2FBC8F",width:"80%"}} >
+      <div style={{padding:"30px",margin:"15px"}} >
+        <input
+          type="text"
+          placeholder="Enter new todo"
+          value={newTodo}
+          onChange={(e) => setNewTodo(e.target.value)}
+        />
+
+        <br/>
+
+        <lable>
+            <DatePicker selected={selectedDate} onChange={date => setDate(date)} customInput={<CustomInput/>} placeholderText="dd/mm/yyyy" />  
+        </lable>
+
+        <br/>
+
+        <button className='add-button' onClick={handleAddTodo}>Add</button>
+      </div>
+      
+      <div >
+      {todos.map((todo) => (
+        <div key={todo.id} className="todo-item" style={{ border: "1px solid black", padding: "10px" }}>
+          {todo.editing ? (
+            <input
+              type="text"
+              value={todo.title}
+              onChange={(e) => handleEditChange(e, todo.id)}
+              onBlur={() => handleUpdate(todo.id)}
+              autoFocus
+            />
+          ) : (
+            <>
+              <input
+                type="checkbox"
+                checked={todo.completed}
+                onChange={() => handleCheckboxChange(todo.id)}
+              />
+              <span style={{ marginRight: "10px" }}>{todo.title}</span>
```

Comment:
the data from your API does not have `title` parameter but rather `description` therefore you cannot see the labels. you should either unify it or you can use `||` condition to allow both variants `{todo.title || todo.descsription}`
![Screenshot 2024-04-04 at 15 37 12](https://github.com/HackYourFuture-CPH/class27-homework/assets/48878547/ab2f4a25-d92e-4d3a-8278-dbb6c749e7fe)